### PR TITLE
Added metrics

### PR DESCRIPTION
**Pull request by l-bat**

### CODE REVIEW

**Comment by l-bat** [comment]
@TolyaTalamanov @KSTconsult  @apertovs @RodionovF

File: src/metrics.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+using namespace deepworks;
+using namespace deepworks::metric;
+
+using ConstMatrix = Eigen::Map<const Eigen::Matrix<float, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>>;
+using ConstVector = Eigen::Map<const Eigen::Matrix<float, 1, Eigen::Dynamic, Eigen::RowMajor>>;
+
+float deepworks::metric::accuracy(const deepworks::Tensor& y_pred,
+                                  const deepworks::Tensor& y_true) {
```
Comment (apertovs) [inline src/metrics.cpp:15]: I think shape size assert will help us in future

**Review by apertovs — APPROVED** [review]
Looks good. I think we should add assert for shapes there.

File: src/metrics.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    const auto& shape = y_pred.shape();
+    int rows = shape[0];
+    int cols = shape[1];
+    assert(y_true.total() == rows);
```
Comment (TolyaTalamanov) [inline src/metrics.cpp:19]: `DeepWorks_Assert` located in `src/util/assert.hpp`

File: src/metrics.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    float acc = 0;
+    ConstMatrix::Index pred_col, label_col;
+    // FIXME: Calculate without loop
+    for (int i = 0; i < rows; i++) {
+        y_pred_mat.row(i).maxCoeff(&pred_col);
+        y_true_mat.row(i).maxCoeff(&label_col);
+        acc += pred_col == label_col;
+    }
+    return acc / rows;
+}
```
Comment (TolyaTalamanov) [inline src/metrics.cpp:53]: New line

File: tests/unit/test_metrics.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    dw::Tensor y_pred(dw::Shape{4, 2});
+    dw::Tensor y_true(dw::Shape{4, 2});
+
+    std::copy(labels.begin(), labels.end(), y_true.data());
+    std::copy(predict.begin(), predict.end(), y_pred.data());
+
+    float expected = 0.5;
+    float acc = dw::metric::sparse_accuracy(y_pred, y_true);
+    EXPECT_FLOAT_EQ(expected, acc);
+}
```
Comment (TolyaTalamanov) [inline tests/unit/test_metrics.cpp:51]: New line

**Review by TolyaTalamanov — APPROVED** [review]

**Comment by TolyaTalamanov** [comment]
LGTM 👍 

Please, resolve conflicts and go merge it

**Review by KSTconsult — APPROVED** [review]

**Pull request merged**